src/order/jordan_holder.lean

```diff
@@ -484,22 +482,22 @@ the least element of `s₁` is the maximum element of `s₂`. -/
 @[simp] lemma append_cast_add {s₁ s₂ : composition_series X}
   (h : s₁.top = s₂.bot) (i : fin s₁.length) :
   append s₁ s₂ h (fin.cast_add s₂.length i).cast_succ = s₁ i.cast_succ :=
-append_cast_add_aux i
+by { dsimp only [append, coe_fn_mk], exact append_cast_add_aux _ _ i, }
```

Comment:
`dsimp` followed by `exact` is a surprising pattern. Is the elaborator having trouble here?

Reply:
If the `dsimp` step is needed, could you turn that into a separate lemma along the lines of `append_apply` and add a docstring explaining why you have to `dsimp` with it?

Reply:
I added the missing lemma, and now can use `rw` instead of `dsimp`.

`eq.symm (append_cast_add_aux _ _ i).symm` also works here as it forces the equality to elaborate in the opposite direction, but that felt too magic!